File: src/main/java/com/example/newstaskapp/view/main/adapters/AllNewsAdapter.java
package com.example.newstaskapp.view.main.adapters;


import static com.example.newstaskapp.view.main.utils.HelperMethod.onLoadImageFromUrl2;
import static com.example.newstaskapp.view.main.utils.HelperMethod.showToast;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.navigation.NavController;
import androidx.recyclerview.widget.RecyclerView;

import com.example.newstaskapp.R;
import com.example.newstaskapp.view.main.data.models.getNewsListResponce.ArticleForRoom;
import com.example.newstaskapp.view.main.views.activities.BaseActivity;

import java.util.List;


public class AllNewsAdapter extends RecyclerView.Adapter<AllNewsAdapter.NewsViewHolder> {
    List<ArticleForRoom> list;
    private final BaseActivity activity;
    private final Context context;

    NavController navController;
//    JoinGroupInterFace joinGroupInterFace;
    public AllNewsAdapter(List<ArticleForRoom> list, NavController navController, Activity activity, Context context) {
        this.list = list;
//        this.joinGroupInterFace=joinGroupInterFace;
        this.navController=navController;
        this.activity = (BaseActivity) activity;
        this.context = context;
    }
    @NonNull
    @Override
    public NewsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.article_card, null, false);
        NewsViewHolder newsViewHolder = new NewsViewHolder(v);
        return newsViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull NewsViewHolder holder, int position) {
//        showToast(activity, list.size()+"");
        setdata(holder,position);
        setaction(holder,position);
    }

    private void setaction(NewsViewHolder holder, int position) {
        holder.cardnewsImgShareBotton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent sendIntent = new Intent();
                sendIntent.setAction(Intent.ACTION_SEND);
                sendIntent.putExtra(Intent.EXTRA_TEXT,
                        list.get(position).getUrl());
                sendIntent.setType("text/plain");
                activity.startActivity(sendIntent);
            }
        });
        holder.cardnewsMainLy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();
                bundle.putSerializable("Articleobj",list.get(position));
                navController.navigate(R.id.nav_show_article_desc, bundle);
            }
        });
    }

    private void setdata(NewsViewHolder holder, int position) {
        ArticleForRoom news = list.get(position);
        try {
            holder.cardnewsDescriptionTv.setText(news.getDescription());
            holder.cardnewsNameTv.setText(news.getName());
            holder.cardnewsTitleTv.setText(news.getTitle());
            if (news.getUrlToImage() != null) {
                String postImage = news.getUrlToImage();
                onLoadImageFromUrl2(holder.cardnewsImg, postImage, context);
            }
        }catch (Exception e){}
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class NewsViewHolder extends RecyclerView.ViewHolder {

        private final View view;
        FrameLayout cardnewsImgShareBotton;
        TextView cardnewsTitleTv;
        TextView cardnewsNameTv;
        TextView cardnewsDescriptionTv;
        ImageView cardnewsImg;
        FrameLayout cardnewsMainLy;
        public NewsViewHolder(@NonNull View itemView) {
            super(itemView);
            view = itemView;
            cardnewsImgShareBotton=view.findViewById(R.id.card_news_item_share_btn);
            cardnewsMainLy=view.findViewById(R.id.card_news_item_framelayout);
            cardnewsImg=view.findViewById(R.id.card_news_item_imageview);
            cardnewsTitleTv=view.findViewById(R.id.card_news_news_items_title_tv);
            cardnewsNameTv=view.findViewById(R.id.card_news_item_name_tv);
            cardnewsDescriptionTv=view.findViewById(R.id.card_news_item_description_tv);

        }
    }
}

File: src/main/java/com/example/newstaskapp/view/main/adapters/AllNewsAdapter.kt
package com.example.newstaskapp.view.main.adapters

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.navigation.NavController
import androidx.recyclerview.widget.RecyclerView
import com.example.newstaskapp.R
import com.example.newstaskapp.view.main.adapters.AllNewsAdapter.NewsViewHolder
import com.example.newstaskapp.view.main.data.models.getNewsListResponce.javaPojo.ArticleForRoom
import com.example.newstaskapp.view.main.utils.HelperMethod
import com.example.newstaskapp.view.main.views.activities.BaseActivity

class AllNewsAdapter(
    var list: List<ArticleForRoom>,
    var navController: NavController,
    activity: Activity,
    context: Context
) :
    RecyclerView.Adapter<NewsViewHolder>() {
    private val activity: BaseActivity
    private val context: Context
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsViewHolder {
        val v =
            LayoutInflater.from(parent.context).inflate(R.layout.article_card, null, false)
        return NewsViewHolder(v)
    }

    override fun onBindViewHolder(holder: NewsViewHolder, position: Int) {
//        showToast(activity, list.size()+"");
        setdata(holder, position)
        setaction(holder, position)
    }

    private fun setaction(holder: NewsViewHolder, position: Int) {
        holder.cardnewsImgShareBotton.setOnClickListener {
            val sendIntent = Intent()
            sendIntent.action = Intent.ACTION_SEND
            sendIntent.putExtra(
                Intent.EXTRA_TEXT,
                list[position].url
            )
            sendIntent.type = "text/plain"
            activity.startActivity(sendIntent)
        }
        holder.cardnewsMainLy.setOnClickListener {
            val bundle = Bundle()
            bundle.putSerializable("Articleobj", list[position])
            navController.navigate(R.id.nav_show_article_desc, bundle)
        }
    }

    private fun setdata(holder: NewsViewHolder, position: Int) {
        val news = list[position]
        try {
            holder.cardnewsDescriptionTv.text = news.description
            holder.cardnewsNameTv.text = news.name
            holder.cardnewsTitleTv.text = news.title
            if (news.urlToImage != null) {
                val postImage = news.urlToImage
                HelperMethod.onLoadImageFromUrl2(holder.cardnewsImg, postImage, context)
            }
        } catch (e: Exception) {
        }
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class NewsViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
        var cardnewsImgShareBotton: FrameLayout
        var cardnewsTitleTv: TextView
        var cardnewsNameTv: TextView
        var cardnewsDescriptionTv: TextView
        var cardnewsImg: ImageView
        var cardnewsMainLy: FrameLayout

        init {
            cardnewsImgShareBotton = view.findViewById(R.id.card_news_item_share_btn)
            cardnewsMainLy = view.findViewById(R.id.card_news_item_framelayout)
            cardnewsImg = view.findViewById(R.id.card_news_item_imageview)
            cardnewsTitleTv = view.findViewById(R.id.card_news_news_items_title_tv)
            cardnewsNameTv = view.findViewById(R.id.card_news_item_name_tv)
            cardnewsDescriptionTv = view.findViewById(R.id.card_news_item_description_tv)
        }
    }

    //    JoinGroupInterFace joinGroupInterFace;
    init {
        //        this.joinGroupInterFace=joinGroupInterFace;
        this.activity = activity as BaseActivity
        this.context = context
    }
}